snuba/query/processors/handled_functions.py

```diff
@@ -0,0 +1,108 @@
+from snuba.clickhouse.columns import ColumnSet
+from snuba.query.validation.signature import SignatureValidator
+from snuba.query.expressions import (
+    Argument,
+    Column,
+    Expression,
+    FunctionCall,
+    Lambda,
+    Literal,
+)
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.query.conditions import (
+    BooleanFunctions,
+    ConditionFunctions,
+    binary_condition,
+)
+from snuba.request.request_settings import RequestSettings
+
+
+class HandledFunctionsProcessor(QueryProcessor):
+    """
+    Adds the isHandled and notHandled snuba functions.
+
+    The implementation of these functions is too complex for clients to provide so
+    these wrappers are required.
+
+    - The `isHandled` function searches an array field for null or 1.
+    - The `notHandled` function searches an array field for 0, null
+      values will be excluded from the result.
+
+    Both functions return 1 or 0 if a row matches.
+    """
+
+    def __init__(self, column: str, columnset: ColumnSet):
+        self.__column = column
+        self.__columnset = columnset
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_functions(exp: Expression) -> Expression:
+            if isinstance(exp, FunctionCall):
+                if exp.function_name == "isHandled":
+                    validator = SignatureValidator([])
+                    validator.validate(exp.parameters, self.__columnset)
```

Comment:
Unfortunately this raises the wrong type of exception.
InvalidFunctionCall does not extend InvalidQueryException. Thus no HTTP400.
This is how the validation logic raises the appropriate exception.
https://github.com/getsentry/snuba/blob/master/snuba/query/parser/validation/functions.py#L51-L57